component/otelcol/processor/resourcedetection/internal/resource_attribute_config/resource_attribute_config.go

```diff
@@ -0,0 +1,12 @@
+package resource_attribute_config
+
+// Configures whether a resource attribute should be enabled or not.
+type ResourceAttributeConfig struct {
+	Enabled bool `river:"enabled,attr"`
+}
```

Comment:
I don't mean to block this PR, since this is already a big effort, but I'm curious, did we consider other possibilities to make this easier to configure and/or maintain in the future?

For example, one possibility was to consider making this a bool type alias, or an enum-like field

So instead of
```
cloud.account.id { enabled = true }
```

We either get
```
cloud.account.id = true
// or
enabled = ["cloud.account.id", "foo", "bar"]
```

Also, AFAICT in the upstream OTEL collector they use some kind of [YAML metadata](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/internal/aws/ec2/metadata.yaml) to generate all these config options, maybe we could use a similar approach?